src/query/storage/m3/storage.go

```diff
@@ -72,19 +73,26 @@ func NewStorage(
 	writeWorkerPool xsync.PooledWorkerPool,
 	tagOptions models.TagOptions,
 	lookbackDuration time.Duration,
-) Storage {
+	conversionCacheSize int,
+) (Storage, error) {
 	opts := m3db.NewOptions().
 		SetTagOptions(tagOptions).
 		SetLookbackDuration(lookbackDuration).
 		SetConsolidationFunc(consolidators.TakeLast)
 
+	conversionLRU, err := storage.NewQueryConversionLRU(conversionCacheSize)
```

Comment:
I think it would make a bit more sense to create the cache above and then pass the cache into the storage constructor?